### PR TITLE
Update 4 for v1.0.3

### PR DESCRIPTION
**Pull request by webadict**
-Added all remaining Old Gods cards... Except Renounce Darkness.
-Fixed a bunch of cards... But I was a little late on a few of them.
-Moved cards to better folders. Some cards will need to have their sets and rarities changed, but that'll be eventually.
-Refactored some spells to work better and broader.

### CODE REVIEW

**Comment by akoscz** [comment]
check out the ValidateCards test to ensure that new cards being added are parseable.
`sh gradlew test -Dtest.single=ValidateCards`

**Comment by webadict** [comment]
I've never been able to get a test working, so I don't know how that would be possible.

**Comment by kairos4242** [comment]
http://www.hearthpwn.com/forums/hearthstone-general/general-discussion/112250-reward-cards-gelbin-old-murk-eye-etc-wont-be

Gelbin Mekkatorque and E.T.C. won't be Standard legal, so they should be either included in the Reward set, or in their own Promo set which is included in Wild but not Standard.

**Comment by webadict** [comment]
Might as well make them their own Promo set as well. Thought I had 'em all.

**Comment by demilich1** [comment]
Just tested this a little bit, Varian Wrynn is throwing Exceptions for me

**Comment by webadict** [comment]
Yeah, my bad. I added a few more changes, and I didn't get to fully test them. Varian Wrynn was a forgotten '!', but I'm checking some other errors.

**Comment by webadict** [comment]
Yo man! I warned you about the weapons! Perdition's Blade is NOT happy! It's trying to check for triggers on old weapons that can't be reached because you forgot to leave a lingering trail! But don't worry, I got it fixed now.

**Comment by demilich1** [comment]
So I added the weapon to the setAside zone too late. My bad. Still, the changes I made to the weapons made the code more clean and were right in general.

**Comment by webadict** [comment]
Yeah, it looks better. I'm surprised that it didn't trigger a death
sequence with King's Defender, but not by much. Silly triggers.

**Comment by webadict** [comment]
HOWEVER, this means that the weapon forces its own death phase later than
it should. Perdition's Blade might need to be tested to determine if this
is true in game, but my guess is that it is not. The death sequence woul
likely follow normal processing, so if a minion was killed at the same
time, whichever was summoned first would process their death first. In this
case, a 4/2 piloted shredder should die if summoned first, I think, since
the battlecry phase shouldn't actually process deaths.

**Comment by webadict** [comment]
Umm... Basically, I'll do some testing. I think Yogg-Saron still needs a bit of work for Discover spells (My guess is that it will make you choose them right now, but it totally shouldn't.) Other than that, it should function correctly.

As for Blood of The Ancient One and Renounce Darkness... That's a bit tougher.

**Comment by webadict** [comment]
Weapons still need to be fixed. Throwing them into the SetAsideZone only makes them findable for triggers and other things. It doesn't allow us to actually access it and kill things from there (Or else lots of bad things could reach us from there!)

So, we need a way to process any deathrattles the replaced weapon might have at the same time it should process other deathrattles from minions.

So, here's another set of Test Cases:
Play Death's Bite.
Play Harvest Golem.
Moonfire Golem.
Play Perdition's Blade, killing Golem.
Since Death's Bite was summoned first, it process death first, dealing 1 damage to everything.
Harvest Golem, now 2/-1, dies as well, summoning the 2/1.

Test Case 2:
Play Harvest Golem,
Play Death's Bite.
Moonfire Golem.
Play Perdition's Blade, killing Golem.
Since Harvest Golem was summoned first, summon the 2/1.
Death's Bite activates, killing the 2/1.
Empty board.

On the plus side, Yogg-Saron is amazingly crazy.

**Comment by demilich1** [comment]
This pull request is getting quite huge. Of course, the reason for that is that you are doing a lot of work on the project and I really appreciate that.

However, for merging and testing it would be better to have multiple smaller pull requests. I just tested the most recent version of your PR and getting exceptions. I would really like to merge that soon, to prevent it from getting bigger and bigger. Could you give me a shout when you consider it stable and ready to be merged?

Also for the future, I would prefer 'focused' pull requests. In the ideal case a PR should change/improve/add one thing and one thing only. Because with that many changes, it is really hard to follow and also makes it impossible to cherry-pick. Let's assume for example I would really dislike the trigger rework (which I don't, just hypothetical): there is no way for me to just exclude that from being merged. I can only reject the whole PR, which means missing out on all the new WOTOG cards and all the other cool stuff you did.
